### PR TITLE
* Graph is implemented to code 4 using real data from RestAPI. Also f…

### PR DESCRIPTION
**Pull request by cazapatamar**
…rist test to graph of code 4 is rendered with real data that is not loaded from RestAPI

### CODE REVIEW

File: src/commons/BarChartData.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        indicatorNameNegative = 'Superficie del área del ecosistema estrategico fuera del bloque';
+        break;
+      default:
+        validIndicator = false;
+        break;
+    }
+
+    return { validIndicator, indicatorNamePositive, indicatorNameNegative };
+  }
+  /**
```
Comment (erikasv) [inline src/commons/BarChartData.js:46]: insert blank line above this line

File: src/commons/BarChartData.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+   * Construct an array to be inserted in the transformed graph data
+   *
+   * @param {Array} rawData
+   * @param {String} valueDescription
+   */
+
```
Comment (erikasv) [inline src/commons/BarChartData.js]: remove this blank line

File: src/commons/BarChartData.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    const rowComputed = [];
+    row.push(valueDescription);
+    rawData.forEach((item) => {
+      if (item.value_description === valueDescription) {
+        row.push(parseFloat(item.indicator_value, 10));
+      }
+    });
+    const desc = row[0];
+    const value = row[1];
+    const valueTransf = (((100 - row[2]) * row[1]) / row[2]);
```
Comment (erikasv) [inline src/commons/BarChartData.js]: The problem with this is that you are assuming that values will always be sorted and the percentage will come later than the area number.

row should be an object where you can be sure which value is which. In that case you need to map somewhere the ids pairs (area value, percentage)

**Review by erikasv — CHANGES_REQUESTED** [review]

File: src/commons/BarChartData.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  static processOneRow(rawData, valueDescription) {
+    const row = {
+      description: valueDescription,
+      value: 0,
+      perc: 0,
+    };
+    const rowComputed = [];
+    rawData.forEach((item) => {
+      if (item.value_description === valueDescription) {
+        if (this.validateValuePercentage(item.id_indicator)) {
```
Comment (erikasv) [inline src/commons/BarChartData.js]: change this for "BarChartData" in every line

File: src/commons/BarChartData.js [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  }
+
+  /**
+   * Validate indicator value vs percentage
+   *
+   * @param {Integer} idIndicator id indicator to be validated
+   */
+  static validateValuePercentage(idIndicator) {
+    let isValue = false;
+    switch (idIndicator) {
```
Comment (erikasv) [inline src/commons/BarChartData.js]: change switch for object ;)

**Review by erikasv — CHANGES_REQUESTED** [review]
Would be better

File: src/commons/BarChartData.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    const row = {
+      description: valueDescription,
+      value: 0,
+      perc: 0,
+    };
+    const rowComputed = [];
+    rawData.forEach((item) => {
+      if (item.value_description === valueDescription) {
+        if (BarChartData.validateValuePercentage(item.id_indicator)) {
+          row.value = parseFloat(item.indicator_value, 10);
```
Comment (alegnaaived) [inline src/commons/BarChartData.js]: Please add numberWithCommas(Number(parseFloat(item.indicator_value, 10)).toFixed(2)) or numberWithCommas(Number(row.value).toFixed(2)) to guarantee the number format to 2 decimals


// Thousands number format
const numberWithCommas = (x) => x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');

**Review by alegnaaived — CHANGES_REQUESTED** [review]

**Review by erikasv — APPROVED** [review]

**Review by alegnaaived — APPROVED** [review]

**Pull request merged**